src/WebJobs.Script.WebHost/WebJobsServiceCollectionExtensions.cs

```diff
@@ -90,13 +89,14 @@ public static IServiceProvider AddWebJobsScriptHost(this IServiceCollection serv
             builder.RegisterType<ScriptEventManager>().As<IScriptEventManager>().SingleInstance();
             builder.Register(c => WebHostSettings.CreateDefault(c.Resolve<ScriptSettingsManager>()));
 
-            // Pass a specially-constructed LoggerFactory to the WebHostResolver. This LoggerFactory is only used
-            // when there is no host available.
+            // Pass a specially-constructed LoggerFactory to the WebHostResolver, if there isn't already a service instance
+            // registered. This LoggerFactory is only used when there is no host available.
             // Only use this LoggerFactory for this constructor; use the registered ILoggerFactory below everywhere else.
+            var defaultLoggerFactory = services.Where(p => p.ServiceType == typeof(ILoggerFactory) && p.ImplementationInstance != null).Select(p => p.ImplementationInstance).FirstOrDefault();
```

Comment:
It is unfortunate we have to do this here. The issue is that we have a circular dependency in our registrations. WebHostResolver depends on ILoggerProvider, but below ILoggerProvider depends on WebHostResolver.

Reply:
This really doesn't look/feel right. Feels like we're getting into some of the old problematic patterns. Is this required for the environment changes or unrelated?

Reply:
Required to support tests that have added a mock ILoggerFactory to the services collection, as the StandbyMode_EndToEnd test does. Without this, the instance I added to the services collection will be ignored and I can't access/verify the test logs.

Open to other ideas on this - I tried several other things, but this is all I could get to work, due to the circular dependency. Basically if an instance of ILoggerFactory has been preconfigured, we want to use that, otherwise create our default.

Reply:
I think what we need is two methods on ILoggerFactoryBuilder. One would add the LoggerProviders for each host (which we have today), and one would add it for the 'fallback', which we do here. Then this logic from CreateLoggerFactory could move there, and a test could implement thta interface to replace the providers in both places.

It shouldn't be a tough change if we agree (I could take it separately from this). It's also a point-in-time issue as our DI work will change how a lot of this registration works.

Reply:
So you suggest I add a new CreateDefaultLoggerFactory method to that interface? I'd then move this default code into there, and I'd call this from within the ctor of WebHostResolver (e.g. `_loggerFactory = loggerProviderFactory.CreateDefaultLoggerFactory(...)`)